### PR TITLE
implement support for merging bgm_property files

### PR DESCRIPTION
**Pull request by jozz024**

### CODE REVIEW

File: src/fs/utils.rs [hunk 1]
```diff
@@ -277,3 +277,46 @@ pub fn add_motionlist_patch<P: AsRef<Path>, Q: AsRef<Path>>(tree: &mut Tree<ApiL
...
+                Err(e) => {
+                    error!("Could not add file {} to API tree. Reason: {:?}", full_path.display(), e);
+                    return None;
+                },
+            }
+        }
+    }
+    error!("Could not add file {} to API tree. Reason: This is not a motion_list.bin file.", full_path.display());
+    None
+}
```
Comment (Raytwo) [inline src/fs/utils.rs:322]: This file is intended for small standalone methods that don't rely on anything else, probably keep this in the same file as the patching like the rest or split them in smaller modules if you find it to be too clogged

File: Cargo.toml [hunk 2]
```diff
@@ -58,6 +58,9 @@ nus3audio = "1.2.0"
 # For motion list patching
 motion_list_rs = "1.4.3"
 serde_yaml = "0.8"
+# for bgm property patching
+smash-bgm-property = "1.0.1"
+binread = "1.4.1"
```
Comment (Raytwo) [inline Cargo.toml:63]: Considering you have a crate specifically to deal with bgm_property, why introduce the dependency just for a trait here?
Offer a ::open method instead on the struct in the crate instead

File: src/fs/loaders.rs [hunk 3]
```diff
@@ -7,6 +7,8 @@ use motion_lib::*;
 use serde::*;
 use serde_xml_rs;
 use serde_yaml::from_str;
+use bgm_property::BgmPropertyFile;
+use binread::BinRead;
```
Comment (Raytwo) [inline src/fs/loaders.rs:11]: Same remark as for Cargo.toml

File: src/fs/utils.rs [hunk 4]
```diff
@@ -277,3 +277,46 @@ pub fn add_motionlist_patch<P: AsRef<Path>, Q: AsRef<Path>>(tree: &mut Tree<ApiL
     error!("Could not add file {} to API tree. Reason: This is not a motion_list.bin file.", full_path.display());
     None
 }
+
+pub fn add_bgm_property_patch<P: AsRef<Path>, Q: AsRef<Path>>(tree: &mut Tree<ApiLoader>, phys_root: P, local: Q) -> Option<Hash40> {
+    let local = local.as_ref();
+    let base_local = local.with_extension("bin");
```
Comment (Raytwo) [inline src/fs/utils.rs:283]: You sure about that? I'd assume there are more files with a .bin extension than bgm_property?

File: src/fs/loaders.rs [hunk 5]
```diff
@@ -325,7 +330,31 @@ impl ApiLoadType {
...
+                let patches = if let Some(patches) = ApiLoader::get_bgm_property_patches_for_hash(local.smash_hash()?) {
+                    patches
+                } else {
+                    return Err(ApiLoaderError::Other("No patches found for file in bgm_property patch!".to_string()));
+                };
+
+                let data = ApiLoader::handle_load_base_file(local)?;
+                let mut reader = std::io::Cursor::new(data);
+
+                let mut bgm_property = BgmPropertyFile::read(&mut reader).unwrap();
```
Comment (Raytwo) [inline src/fs/loaders.rs:344]: About the "open" method, check this usage in [smash-arc for example](https://github.com/jam1garner/smash-arc/blob/master/src/arc_file.rs#L154)

**Review by Raytwo — CHANGES_REQUESTED** [review]

File: src/fs/utils.rs (hunk 4 above)
Comment (jozz024) [reply src/fs/utils.rs:283]: messaged you about this already, but with_local just sets the extension to the extension given, so i'm not sure what you mean by this.

**Review by Raytwo — APPROVED** [review]

**Pull request merged**